src/agl/WebRuntimeAGL.cpp

```diff
@@ -265,6 +258,9 @@ bool WebAppLauncherRuntime::init() {
       m_role = "homescreen";
     else if (m_id.rfind("webapps-homescreen", 0) == 0)
       m_role = "homescreen";
+    else {
+      m_role = m_id.substr(0,12);
```

Comment:
Application ID is a bad idea for this purpose, as with the typical naming it is still easy to see duplicates. I.e. com.webos.app. will alll have the same role.

I think we can take a different solution, that is just applying a hash algorithm. I.e. wapp-XXXXXXX (where XXXXXXX is calculated from the application ID, that is expected to be unique).

Another possibility is requiring apps to expose their roles somehow in config.xml.

And another possibility is fixing the platform to not fail with a role that is bigger than 12 characters. Do we know where this limit comes from?

Reply:
> ...
> I think we can take a different solution, that is just applying a hash algorithm. I.e. wapp-XXXXXXX (where XXXXXXX is calculated from the application ID, that is expected to be unique).

Thanks, will try this.

> Another possibility is requiring apps to expose their roles somehow in config.xml.

In further versions of AGL, a new Wayland compositor will be used that replaces current homescreen and windowmanager APIs. Role names as they are now are expected to disappear, so it's not worth proposing changes in the application manifest, IMHO.

> And another possibility is fixing the platform to not fail with a role that is bigger than 12 characters. Do we know where this limit comes from?

This patch is intended as a workaround while the actual problem is not fixed, to reach the release date.